modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/filehandle/FileWriteHandleImpl.java

```diff
@@ -500,7 +500,19 @@ private void fsync(MappedByteBuffer buf, int off, int len) throws IgniteCheckedE
 
                             switchSegmentRecordOffset = filePtr.fileOffset() + switchSegmentRecSize;
                         }
+                        else {
+                            if (log.isDebugEnabled())
+                                log.debug("Not enough space in wal segment to write segment switch");
+                        }
                     }
+                    else {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Not enough space in wal segment to write segment switch, written="
+                                + written + ", switchSegmentRecSize=" + switchSegmentRecSize);
+                        }
+                    }
+
+                    flushOrWait(null);
```

Comment:
Please indicate why there is **null**.

Reply:
null is for unconditional flush, the same as 20  lines above